docs/adr/2022-12-05_021-Bounded-transaction-validity-on-Hydra-protocol-transactions.md

```diff
@@ -0,0 +1,102 @@
+---
+slug: 21
+title: |
+  21. Bounded transaction validity on Hydra protocol transactions
+authors: [ch1bo, Pascal, Franco, Arnaud, v0d1ch]
+tags: [Accepted]
+---
+
+## Status
+
+Proposed
+
+## Context
+
+- The HydraHeadV1 formal specification contains a _bounded confirmation window_:
+
+  ```
+  // Deadline
+
+  T_max <= T_min + L // Bounded confirmation window
+  DL’ = T_max + L    // The latest possible deadline is 2*L
+
+  ```
+
+  with `T_min` and `T_max` being the tx validity bounds and `L` being the
+  contestation period.
+
+  - This is to avoid attacks with specified upper validity bound being too far
+    in the future and denial of service the head with this (e.g. 10 years).
+
+#### Current state of things:
+
+- The contestation period and upper tx validity is used for computing the
+  contestation deadline.
+
+- There is a `closeGraceTime` currently hard-coded (to `100` slots) to set some
+  upper bound on the `closeTx`. This was also required so far to compute the
+  contestation deadline.
+
+- Different networks (chains) have different slot lenghts, e.g. the preview
+  network has a slot every `1s`, while our local devnets use `0.1s`. This means
+  hardcoded values like `closeGraceTime` need to be _in sync_ with the
+  underlying network.
+
+- The `contestationPeriod` can be configured by users via the `Init` client
+  input. For example, the hydra-cluster test suite uses a hardcoded `cperiod` on
+  the client side.
+
+- Default value for `T_Min` is negative infinity.
+
+- Lower tx validity being in the future does not pose a problem since other
+  participant is able to close a head.
+
+#### What we want to achieve:
+
+- We want to enforce topmost formula in this file in our code on-chain.
+
+- Introduce `maxGraceTime` expressed in seconds in place of `closeGraceTime` and adjust to
+  appropriate value.
+
+- The upper tx validity of close transaction should be minimum between
+  `maxGraceTime` and `contestationPeriod`. Users need to know this.
```

Comment:
Is this now context, decision or consequence? It's mentioned everywhere :thinking:

Reply:
I'd say it's really a consequence of the fact that we cannot convert arbitrary point in times to slots (which is required by the ledger when constructing transactions).